site/docs/references/custom-selectors.md

```diff
@@ -0,0 +1,164 @@
+---
+id: references/custom-selectors
+title: Custom Selectors
+layout: docs
+---
+
+You use [Custom selectors](https://drafts.csswg.org/css-extensions/#custom-selectors) to define a an alias name matching complex selectors.
+
+## Usage
+
+The following code maps the alias name `controlBtn` that matches any `.btn` CSS class nested under the `.controls` CSS class:
+
+**CSS API**
+```css
+@namespace "Comp";
+@custom-selector :--controlBtn .controls .btn;
+/*
+selector: .Comp__root .Comp__controls .Comp__btn
+*/
+:--controlBtn { border: 1px solid grey; }
+/*
+selector: .Comp__root .Comp__controls .Comp__btn:hover
+*/
+:--controlBtn:hover { border-color: red; }
+```
+
+### Expose pseudo-element
+
+Custom selectors generate a [pseudo-element](./pseudo-elements.md), So for example [importing](./imports.md) the previous stylesheet into another stylesheet allows access to the `form-input` pseudo-element:
+
+```css
+@namespace "Page";
+:import {
+    -st-from: "./comp.st.css";
+    -st-default: Comp;
+}
+/*
+selector: .Page__root .Comp__root .Comp__controls .Comp__btn
+*/
+Comp::controlBtn { 
+    background: gold; 
+}
+```
+
+> **Notice**:  
+> In case a `custom-selector` alias conflicts with a local CSS class name, the exposed `pseudo-element` will target the `custom-selector`. However the the exported CSS class will still be exported to Javascript.
+
+## Use cases
+
+### Container and recursive components
+
+Some components might contain nested instances of themselves, because they're a container or a "recursive" component (e.g. Tree component might render itself). 
+
+In case the component exposes any `pseudo-elements`, it is a good practice to define them as `custom-selectors` with [child selectors](https://developer.mozilla.org/en-US/docs/Web/CSS/Child_selectors) to avoid effecting internal instances inner parts.
+
+The following example show how a tree component exposes an icon:
+
+```css
+@namespace "Tree";
+@custom-selector :--icon .root > .icon;
+```
+
+Using the icon `pseudo-selector` from the outside just like a CSS class selector:
+```css
+@namespace "Panel";
+:import {
+    -st-from: "./tree.st.css";
+    -st-default: Tree;
+}
+/*
+selector: .Panel__root .Tree__root > .Tree__icon
+*/
+Tree::icon {
+    background: yellow;
+}
+```
+
+### Expose deep inner parts
+
+When you want to make some internal parts more accessible in your component API, you can simply describe `pseudo-elements` using `custom selectors`.
+
+For example exposing a `pseudo-element` named `navigationBtn` that allow to style an internal gallery component `navBtn` element:
+```css
+:import {
+    -st-from: "./gallery.st.css";
+    -st-default: Gallery;
+}
+@custom-selector :--navigationBtn Gallery::navBtn;
+```
+
+### Combination selector
+
+Sometimes a component has several basic CSS classes (with corresponding `pseudo-elements`) and we want to expose a combination `pseudo-element`.
```

Comment:
i don't understand this premise? are we just connecting two classes with a single selector?